src/cpu/core_dyn_x86/cache.h

```diff
@@ -126,11 +127,14 @@ class CodePageHandler : public PageHandler {
 		return map;
 	}
 
-	void writeb(PhysPt addr,Bitu val){
-		if (GCC_UNLIKELY(old_pagehandler->flags&PFLAG_HASROM)) return;
-		if (GCC_UNLIKELY((old_pagehandler->flags&PFLAG_READABLE)!=PFLAG_READABLE)) {
-			E_Exit("wb:non-readable code page found that is no ROM page");
-		}
+	// the following functions will clean all cache blocks that are invalid
+	// now due to the write
+
+	void writeb(PhysPt addr, Bitu val)
+	{
+		assert((old_pagehandler->flags & PFLAG_HASROM) == 0x0);
+		assert(old_pagehandler->flags & PFLAG_READABLE);
```

Comment:
Thanks for the excellent comment, and an approach that gives us a safety net if this check is actually needed.

Reply:
There are plenty of helpful comments, that someone added when turning core_dyn_x86/cache.h into core_dynrec/cache.h years ago. I think I'll do unification by copying comments in the future PR.

No idea about plan for removing these asserts though. Hopefully we won't need to turn them again into if-statements…